src/ConnectedMode/BranchMatcher.cs

```diff
@@ -35,10 +35,11 @@ public interface IBranchMatcher
     {
         /// <summary>
         /// Calculates the Sonar server branch that is the closest match to the head branch.
-        /// Defaults to branch marked as "main" when no match is found or when there are no head branches.
+        /// Returns null if there are no head branches
         /// </summary>
         /// <remarks>
         /// Compares branches by name and calculates distances by the number of different commits.
+        /// Defaults to branch marked as "main" when no match found.
```

Comment:
Looks like the behaviour is to return `null` for detached heads
i.e. it returns `null` for detached heads, the closest match if it could find one, or `main` if it couldn't.

If the caller now has to handle `null` (and has the logic to find `main` if the branch matcher returns `null`), this class could be simpler: return closest match, or null in all other cases.

Reply:
I'll open a separate issue for it 👍